system/nxcamera/nxcamera.c

```diff
@@ -0,0 +1,889 @@
+   * to perform clean-up.
+   */
+
+  if (pcam->loop_id != 0)
+    {
+      pthread_join(pcam->loop_id, &value);
+    }
+
+  pthread_attr_init(&tattr);
+  sparam.sched_priority = sched_get_priority_max(SCHED_FIFO) - 9;
+  pthread_attr_setschedparam(&tattr, &sparam);
+  pthread_attr_setstacksize(&tattr, CONFIG_NXCAMERA_LOOPTHREAD_STACKSIZE);
+
+  /* Add a reference count to the looper for the thread and start the
+   * thread.  We increment for the thread to avoid thread start-up
+   * race conditions.
+   */
+
+  nxcamera_reference(pcam);
+  ret = pthread_create(&pcam->loop_id, &tattr, nxcamera_loopthread,
+                       (pthread_addr_t)pcam);
+  pthread_attr_destroy(&tattr);
+  if (ret != OK)
+    {
+      ret = -ret;
+      verr("ERROR: Failed to create loopthread: %d\n", ret);
+      goto err_out;
+    }
+
+  /* Name the thread */
+
+  pthread_setname_np(pcam->loop_id, "nxcameraloop");
+  return OK;
+
+err_out:
+  if (pcam->bufs)
+    {
+      for (i = 0; i < pcam->nbuffers; i++)
+        {
+          if (pcam->bufs[i] != NULL && pcam->bufs[i] != MAP_FAILED)
+            {
+              munmap(pcam->bufs[i], pcam->buf_sizes[i]);
+            }
+        }
+
+      free(pcam->bufs);
+    }
+
+  if (pcam->buf_sizes)
+    {
+      free(pcam->buf_sizes);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: nxcamera_create
+ *
+ *   nxcamera_create() allocates and initializes a nxcamera context for
+ *   use by further nxcamera operations.  This routine must be called before
+ *   to perform the create for proper reference counting.
+ *
+ * Input Parameters:  None
+ *
+ * Returned values:
+ *   Pointer to the created context or NULL if there was an error.
+ *
+ ****************************************************************************/
+
+FAR struct nxcamera_s *nxcamera_create(void)
+{
+  FAR struct nxcamera_s *pcam;
+  int err;
+
+  /* Allocate the memory */
+
+  pcam = (FAR struct nxcamera_s *)calloc(1, sizeof(struct nxcamera_s));
+  if (pcam == NULL)
+    {
+      return NULL;
+    }
+
+  /* Initialize the context data */
+
+  pcam->loopstate = NXCAMERA_STATE_IDLE;
+  pcam->display_fd = -1;
+  pcam->capture_fd = -1;
+  err = pthread_mutex_init(&pcam->mutex, NULL);
+  if (err)
+    {
+      verr("ERROR: pthread_mutex_init failed: %d\n", err);
+      free(pcam);
+      pcam = NULL;
+    }
+
+  return pcam;
+}
+
+/****************************************************************************
+ * Name: nxcamera_release
+ *
+ *   nxcamera_release() reduces the reference count by one and if it
+ *   reaches zero, frees the context.
+ *
+ * Input Parameters:
+ *   pcam    Pointer to the NxCamera context
+ *
+ * Returned values:    None
+ *
+ ****************************************************************************/
+
+void nxcamera_release(FAR struct nxcamera_s *pcam)
+{
+  int      refcount;
+  FAR void *value;
+  int errcode;
```

Comment:
```suggestion
  int      refcount;
  FAR void *value;
  int      errcode;
```
or remove spaces before `refcount`